generic/ci/security/use-frozen-lockfile.yaml

```
@@ -1,19 +1,29 @@
 rules:
 - id: use-frozen-lockfile-yarn
   patterns:
-  - pattern-regex: yarn install\b
-  - pattern-not-regex: yarn install --frozen-lockfile
-  - pattern-not-regex: yarn install --immutable
-  - pattern-not-regex: yarn install -g
-  - pattern-not-regex: yarn install --global
-  - pattern-not-regex: yarn install [\w]+
-  fix: yarn install --frozen-lockfile
+    - pattern: |
+        RUN ... yarn $INSTALL ...
+    - pattern-not-inside: |
+        RUN ... yarn $INSTALL ... --frozen-lockfile ... 
+    - pattern-not-inside: |
+        RUN ... yarn $INSTALL ... --immutable ...
+    - pattern-not-inside: |
+        RUN ... yarn $INSTALL ... -g ...
+    - pattern-not-inside: |
+        RUN ... yarn $INSTALL ... --global ...
+    - metavariable-regex: 
+        metavariable: $INSTALL
+        regex: (install)
```

```suggestion
        regex: ^(install)$
```
Should we anchor this to ensure that there won't be false positives for other commands that use a word containing the word "install" in it?